src/OctoshiftCLI.Tests/ado2gh/Commands/InventoryReportCommandTests.cs

```diff
@@ -135,10 +136,40 @@ public async Task It_Uses_The_Ado_Pat_When_Provided()
             await _command.Invoke("some org", adoPat);
 
             _mockAdoApiFactory.Verify(m => m.Create(adoPat));
-            _mockOrgsCsvGenerator.Verify(m => m.Generate(adoPat));
-            _mockTeamProjectsCsvGenerator.Verify(m => m.Generate(adoPat));
-            _mockReposCsvGenerator.Verify(m => m.Generate(adoPat));
+            _mockOrgsCsvGenerator.Verify(m => m.Generate(adoPat, It.IsAny<bool>()));
+            _mockTeamProjectsCsvGenerator.Verify(m => m.Generate(adoPat, It.IsAny<bool>()));
+            _mockReposCsvGenerator.Verify(m => m.Generate(adoPat, It.IsAny<bool>()));
             _mockPipelinesCsvGenerator.Verify(m => m.Generate(adoPat));
         }
+
+        [Fact]
+        public async Task It_Generates_Minimal_Csvs_When_Requested()
+        {
+            // Arrange
+            var expectedOrgsCsv = "csv stuff";
+            var expectedTeamProjectsCsv = "more csv stuff";
+            var expectedReposCsv = "repo csv stuff";
+            var expectedPipelinesCsv = "pipelines csv stuff";
+
+            _mockAdoApiFactory.Setup(m => m.Create(null)).Returns(_mockAdoApi.Object);
+
+            _mockOrgsCsvGenerator.Setup(m => m.Generate(null, It.IsAny<bool>())).ReturnsAsync(expectedOrgsCsv);
+            _mockTeamProjectsCsvGenerator.Setup(m => m.Generate(null, It.IsAny<bool>())).ReturnsAsync(expectedTeamProjectsCsv);
+            _mockReposCsvGenerator.Setup(m => m.Generate(null, It.IsAny<bool>())).ReturnsAsync(expectedReposCsv);
+            _mockPipelinesCsvGenerator.Setup(m => m.Generate(null)).ReturnsAsync(expectedPipelinesCsv);
+
+            // Act
+            await _command.Invoke(null, minimal: true);
+
+            // Assert
+            _orgsCsvOutput.Should().Be(expectedOrgsCsv);
+            _teamProjectsCsvOutput.Should().Be(expectedTeamProjectsCsv);
+            _reposCsvOutput.Should().Be(expectedReposCsv);
+            _pipelinesCsvOutput.Should().Be(expectedPipelinesCsv);
+
+            _mockOrgsCsvGenerator.Setup(m => m.Generate(null, true));
```

Comment:
shouldn't these be .Verify() ? alternatively you could just make the setup above more specific

Reply:
Great catch! they should have been all `Verify()`s. In this particular test, I prefer to explicitly use `Verify`s in the end rather than implicitly verify as part of the `Setup` code because those `Verify`s are actually what I want to test so I think it's more clear to have them in the `Assert` section.